packages/block-library/src/footnotes/format.js

```diff
@@ -62,6 +63,16 @@ export const format = {
 			return null;
 		}
 
+		// Checks if the selected block lives within a pattern.
+		if (
+			getBlockParentsByBlockName(
```

Comment:
We shouldn't be calling selectors like that in a component render call. This is only fine in event callbacks. We should create a separate `useSelect` call that listens for changes to the store so the component re-renders when the result of this function changes.

Reply:
Oh good one, thanks for the nudge.

I've done this, slightly verbosely... open to optimization recommendations 😄